controller_manager/scripts/spawner.py

```diff
@@ -0,0 +1,138 @@
+                          '--controller-manager', controller_manager_name], capture_output=True,
+                         encoding='utf8')
+    output = str(ret.stdout)
+    for line in output.splitlines():
+        if controller_name in line.split('[')[0]:
+            return True
+    return False
+
+
+def main(args=None):
+
+    rclpy.init(args=args)
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'controller_name', help='Name of the controller')
+    parser.add_argument(
+        '-t', '--controller-type',
+        help='If not provided it should exist in the controller manager namespace',
+        default=None, required=False)
+    parser.add_argument(
+        '-c', '--controller-manager', help='Name of the controller manager ROS node',
+        default='/controller_manager', required=False)
+    parser.add_argument(
+        '-p', '--param-file', help='Controller param file to be loaded into controller '
+        'node before configure', required=False)
+    parser.add_argument(
+        '-u', '--unload-on-kill',
+        help='Wait until this application is interrupted and unload controller',
+        action='store_true')
+
+    command_line_args = rclpy.utilities.remove_ros_args(args=sys.argv)[1:]
+    args = parser.parse_args(command_line_args)
+    controller_name = args.controller_name
+    controller_manager_name = args.controller_manager
+    param_file = args.param_file
+    controller_type = args.controller_type
+
+    node = Node('spawner_' + controller_name)
+    try:
+
+        # Wait for controller_manager
+        timeout = node.get_clock().now() + Duration(seconds=10)
+        while node.get_clock().now() < timeout:
+            ret = subprocess.run(
+                ['ros2', 'service', 'type',
+                 '/' + controller_manager_name + '/load_and_start_controller'],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL)
+            if ret.returncode == 0:
+                break
+            node.get_logger().info(
+                'Waiting for {} services'.format(controller_manager_name),
+                throttle_duration_sec=2)
+            time.sleep(0.2)
+
+        if controller_type:
+            ret = subprocess.run(['ros2', 'param', 'set', controller_manager_name,
+                                  controller_name + '.type', controller_type])
+
+        if is_controller_loaded(controller_manager_name, controller_name):
+            node.get_logger().info('Controller already loaded, skipping load_controller')
+        else:
+            ret = subprocess.run(['ros2', 'control', 'load_controller', controller_name,
+                                  '--controller-manager', controller_manager_name])
+            if ret.returncode != 0:
+                # Error message printed by ros2 control
+                return ret.returncode
+            node.get_logger().info('Loaded ' + controller_name)
+
+        if param_file:
```

Comment:
will this also be executed if the controller is loaded? I this intended?

Reply:
Yes, I believe that if you are calling this script you expect that at the end of it you have a loaded + configured + started controller, and if you provided a parameter file, it should be loaded. 

So when you are testing parameter, you would stop the controller and call this script to load params, configure and start.